src/corehost/corehost.cpp

```diff
@@ -338,5 +353,39 @@ int main(const int argc, const pal::char_t* argv[])
         trace::info(_X("}"));
     }
 
-    return run(argc, argv);
+#if defined(_WIN32) && defined(FEATURE_APPHOST)
+    // Detect if the process has console
+    // Note that this will fail even if a GUI process has STDOUT redirected. So this pretty effectively detects
+    // if the process is a GUI process without a console window.
+    HANDLE con_out = ::CreateFileW(L"CONOUT$", GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_EXISTING, 0, 0);
```

Comment:
Do we need to close this handle when we open it successfully?

Reply:
Good point - we probably should. We're opening it for full sharing, so in theory it should not matter much, but it's definitely cleaner.